stdlib/src/builtin/string.mojo

```diff
@@ -1406,6 +1412,73 @@ struct String(
             substr._strref_dangerous(), start=start
         )
 
+    fn isspace[str_len: Int = -1](self) -> Bool:
```

Comment:
The default is detemining it at runtime, I wanted to leave and overload for when we start checking for longer utf8 sequences for other methods like `splitlines()` and `split()` WDYT?

Reply:
As I said, we can think about optimizing this later, but in this patch this is premature. It would be better to have small patches, that do fewer things; makes it much easier to review and land them internally.